galley/pkg/resource/rewrite.go

```diff
@@ -0,0 +1,44 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"istio.io/istio/galley/pkg/common"
+)
+
+// rewrite does a generic rewrite of an unstructured custom resource.
+func rewrite(source *unstructured.Unstructured, apiVersion string) *unstructured.Unstructured {
```

Comment:
The name of this function could be more descriptive. Maybe `rewriteAPIVersion`?

Reply:
It doesn't just rewrite version. It rewrites the resource (hence resource.rewrite).

If we need to update the spec, this is where we would do it.